src/raw-value.ts

```diff
@@ -126,64 +78,136 @@ function createReverseShadowTarget(target: ReverseProxyTarget): ReverseShadowTar
-                const originalDescriptor = getReverseDescriptor(descriptors[key]);
-                installDescriptorIntoShadowTarget(shadowTarget, key, originalDescriptor);
+    function getSecurePartialDescriptor(rawPartialDesc: PropertyDescriptor): PropertyDescriptor {
+        const secPartialDesc = assign(ObjectCreate(null), rawPartialDesc);
+        if ('value' in secPartialDesc) {
+            // we are dealing with a value descriptor
+            secPartialDesc.value = env.getSecureValue(secPartialDesc.value);
+        }
+        if ('set' in secPartialDesc) {
+            // we are dealing with accessors
+            secPartialDesc.set = env.getSecureValue(secPartialDesc.set);
+        }
+        if ('get' in secPartialDesc) {
+            secPartialDesc.get = env.getSecureValue(secPartialDesc.get);
+        }
+        return secPartialDesc;
+    }
+
+    function lockShadowTarget(shadowTarget: ReverseShadowTarget, originalTarget: ReverseProxyTarget) {
+        // todo: leaking symbols
+        const targetKeys = ownKeys(originalTarget);
+        for (let i = 0, len = targetKeys.length; i < len; i += 1) {
+            const key = targetKeys[i];
+            const rawDesc = ReflectGetOwnPropertyDescriptor(shadowTarget, key);
+            if (isUndefined(rawDesc) || rawDesc.configurable === true) {
+                const secDesc = ReflectGetOwnPropertyDescriptor(originalTarget, key) as PropertyDescriptor;
+                ReflectDefineProperty(shadowTarget, key, getRawDescriptor(secDesc));
             }
         }
+        ReflectPreventExtensions(shadowTarget);
     }
 
     class ReverseProxyHandler implements ProxyHandler<ReverseProxyTarget> {
         // original target for the proxy
         private readonly target: ReverseProxyTarget;
-        // metadata about the shape of the target
-        private readonly meta: TargetMeta;
 
-        constructor(target: ReverseProxyTarget, meta: TargetMeta) {
+        constructor(target: ReverseProxyTarget) {
             this.target = target;
-            this.meta = meta;
-        }
-        initialize(shadowTarget: ReverseShadowTarget) {
-            const { meta } = this;
-            // adjusting the proto chain of the shadowTarget (recursively)
-            const rawProto = env.getRawValue(meta.proto);
-            ReflectSetPrototypeOf(shadowTarget, rawProto);
-            // defining own descriptors
-            copyReverseOwnDescriptors(shadowTarget, meta.descriptors);
-            // reserve proxies are always frozen
-            freeze(shadowTarget);
             // future optimization: hoping that proxies with frozen handlers can be faster
             freeze(this);
         }
-        apply(shadowTarget: ReverseShadowTarget, thisArg: RawValue, argArray: RawValue[]): RawValue {
+        get(shadowTarget: ReverseShadowTarget, key: PropertyKey, receiver: RawObject): SecureValue {
+            const { target } = this;
+            // todo: leaking key as symbol
```

Comment:
What's this todo about ☝️

Reply:
https://github.com/caridy/secure-javascript-environment/issues/60

Reply:
now that #60 is not an issue anymore, I will clean this up.